src/org/rascalmpl/parser/gtd/stack/AbstractStackNode.java

```diff
@@ -668,103 +679,168 @@ public int updateOvertakenNullableNode(AbstractStackNode<P> predecessor, Abstrac
 		}else{ // The predecessor has prefixes.
 			for(int i = edgesMapToAdd.size() - 1; i >= 0; --i){
 				int locationStart = edgesMapToAdd.getKey(i);
-				
+
 				// Prefix not present, add it.
 				int index = edgesMap.findKey(locationStart);
-				
+
 				ArrayList<Link> prefixes;
 				if(index == -1){ // No prefix set for the given start location is present yet.
 					index = edgesMap.size();
 					edgesMap.add(locationStart, edgesMapToAdd.getValue(i));
 					propagatedPrefixes.set(index);
-					
+
 					prefixes = new ArrayList<>(1);
 					prefixesMap[index] = prefixes;
-					
+
 					++nrOfAddedEdges;
 				}else{ // A prefix set for the given start location is present.
 					if(propagatedPrefixes.isSet(index)) continue; // Prefix present, abort.
-					
+
 					prefixes = prefixesMap[index];
 				}
-				
+
 				// Add the prefix to the prefix set.
 				prefixes.add(new Link(prefixesMapToAdd[i], result));
 			}
 		}
-		
+
 		return nrOfAddedEdges;
 	}
-	
+
 	/**
 	 * This method is a specialized version of 'updateNode'.
-	 * 
+	 *
 	 * When alternatives have a shared prefix, their successors can shared the same edges and prefixes maps.
 	 */
 	public void updatePrefixSharedNode(IntegerObjectList<EdgesSet<P>> edgesMap, ArrayList<Link>[] prefixesMap){
 		this.edgesMap = edgesMap;
 		this.prefixesMap = prefixesMap;
 	}
-	
+
 	/**
 	 * Returns the edges map.
 	 */
 	public IntegerObjectList<EdgesSet<P>> getEdges(){
 		return edgesMap;
 	}
-	
+
 	/**
 	 * Returns the incoming edges set.
 	 */
 	public EdgesSet<P> getIncomingEdges(){
 		return incomingEdges;
 	}
-	
+
 	/**
 	 * Returns the prefixes map.
 	 */
 	public ArrayList<Link>[] getPrefixesMap(){
 		return prefixesMap;
 	}
-	
+
 	/**
 	 * Returns the list of start location for which the results have already
 	 * been propagated (hidden-right-recursion specific).
 	 */
 	public IntegerList getPropagatedReductions(){
 		if(propagatedReductions == null) propagatedReductions = new IntegerList();
-		
+
 		return propagatedReductions;
 	}
-	
+
+	public String toShortString() {
+		return null;
+	}
+
+	@Override
+	public String toString() {
+		StringBuilder builder = new StringBuilder();
```

Comment:
Nit: why is this not a big string concat? does the `StringBuilder` improve the readability? If it's for performance, the java compiler automatically translate large string concats to calls to `StringBuilder`.

Reply:
When I wrote this to use "+" and "+=", I got a warning from sonarlint: "Use a StringBuilder instead" :-(